### ISSUE TITLE
[0.74.0-rc.5] - Memory leak when using fetch with new arch

### ISSUE THREAD

**Issue by migueldaipre** [facebook/react-native#43647]
### Description

Use the Fetch API and see memory consumption increase without the automatic cleaning of the garbage collector.

Related #42374 #39100 https://github.com/facebook/hermes/issues/1147

### Steps to reproduce

1. Open Xcode
2. yarn install
3. yarn ios
4. Click on start
5. See the memory profile as in the demo video in the player

### React Native Version

0.74.0-rc.5

### Affected Platforms

Runtime - iOS

### Areas

TurboModule - The New Native Module System

### Output of `npx react-native info`

```text
System:
  OS: macOS 14.3
  CPU: (8) arm64 Apple M1
  Memory: 103.22 MB / 8.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.11.0
    path: ~/.nvm/versions/node/v20.11.0/bin/node
  Yarn:
    version: 3.6.4
    path: ~/.nvm/versions/node/v20.11.0/bin/yarn
  npm:
    version: 10.2.4
    path: ~/.nvm/versions/node/v20.11.0/bin/npm
  Watchman:
    version: 2024.01.22.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.14.3
    path: /Users/migueldaipre/.rbenv/shims/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.2
      - iOS 17.2
      - macOS 14.2
      - tvOS 17.2
      - visionOS 1.0
      - watchOS 10.2
  Android SDK:
    API Levels:
      - "33"
      - "34"
    Build Tools:
      - 30.0.3
      - 33.0.0
      - 33.0.1
      - 34.0.0
    System Images:
      - android-33 | Google APIs ARM 64 v8a
      - android-34 | Google Play ARM 64 v8a
    Android NDK: Not Found
IDEs:
  Android Studio: 2023.1 AI-231.9392.1.2311.11330709
  Xcode:
    version: 15.2/15C500b
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.10
    path: /usr/bin/javac
  Ruby:
    version: 3.3.0
    path: /Users/migueldaipre/.rbenv/shims/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.74.0-rc.5
    wanted: 0.74.0-rc.5
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: true
```

### Stacktrace or Logs

```text
no crash log. Run the reproduction demo, if you watch enough time, you will see it.
```

### Reproducer

https://github.com/migueldaipre/RN74FetchLeak

### Screenshots and Videos

https://github.com/facebook/react-native/assets/21174859/02dcdbdc-5edf-421c-b4b9-429ce55e2163

![print](https://github.com/facebook/react-native/assets/21174859/4b3ba060-5091-405b-93aa-a33c773e5c86)

**Comment by cortinico** [comment]
> ###
> TurboModule - The New Native Module System

Is this related to the New Architecture? 
Seems like a duplicate of https://github.com/facebook/react-native/issues/42374 https://github.com/facebook/react-native/issues/39100 no?

**Comment by migueldaipre** [comment]
> > TurboModule - The New Native Module System
> 
> Is this related to the New Architecture? Seems like a duplicate of #42374 #39100 no?

Yes, this is happening with the old and the new arch.

Could you help us find a fix for the problem? 

In apps that make many requests we have out of memory problems.

**Comment by manuelson** [comment]
hello we got same here

**Comment by migueldaipre** [comment]
> hello we got same here on Android

Both platforms have been fixed in [0.75-rc3](https://github.com/facebook/react-native/blob/main/CHANGELOG.md#v0750-rc3)
